README.md

```diff
@@ -1,2 +1,89 @@
-# docker-calico-route-reflector-init
-Prepares a node to become a route reflector.
+# Calico Route Reflector Init
+
+An [init container](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/) that configures a Kubernetes node to act as a Calico [BGP route reflector](https://docs.projectcalico.org/v3.9/networking/routereflector#content-main).
+
+## Overview
+
+The container performs two tasks that automates the configuration of a node for route reflection:
+
+1. Applies a [label](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/) to the node which should be used in [`BGPPeer`](https://docs.projectcalico.org/v3.9/reference/resources/bgppeer#bgp-peer-definition) resources to indicate which nodes should (or should not) peer with this node.
+2. Applies an [annotation](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/) to the node specifying a cluster identifier, indicating that this node is a route reflector.
+
+
+## Usage
+
+This should be used as an [init container](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/) for the `calico-node` daemonset that will be acting as both a `calico-node` and a route reflector, thus we recommend two sepearate daemonsets of `calico-node`: 
+* One that  targets nodes intended to be route reflectors (i.e. masters).
+* One that targets the rest of the nodes (i.e. workers).
+
+### Example
+
+In this example, we're configuring the masters to be our route reflectors.
+
+```yaml
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: calico-node-master
+  namespace: kube-system
+spec:
+  ...
+  template:
+    ...
+    spec:
+      ...
+      nodeSelector:
+        # Run calico-node-master on masters only.
+        node-role.kubernetes.io/master: ""
+      tolerations:
+        - key: node-role.kubernetes.io/master
+          effect: NoSchedule
+      initContainers:
+        - name: configure-route-reflector
+          image: zappi/calico-route-reflector-init:0.1.0
+          env:
+            # Use the Downward API set the node name.
+            - name: KUBERNETES_NODE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+            - name: CALICO_ROUTE_REFLECTOR_LABEL
+              value: "kops.k8s.io/calico-route-reflector"
+            - name: CALICO_ROUTE_REFLECTOR_CLUSTER_ID
+              value: "1.1.1.1"
```

Comment:
Where do we get this from?

Reply:
The [documentation](https://docs.projectcalico.org/v3.9/reference/resources/node#bgp) around it isn't very clear on what to set it to, probably because its mostly a BGP-specific term and they assume some knowledge on it.

From https://docs.projectcalico.org/v3.9/networking/bgp#configuring-in-cluster-route-reflectors:
> **Note**: For a simple deployment, all the route reflector nodes should have the same cluster ID.

From https://octetz.com/posts/rr-setup#configuring-nodes:
> The `routeReflectorClusterID` represents the BGP router’s cluster id, this is **not** synonymous with a Kubernetes cluster identifier. Determining whether reflectors should share cluster ids may depend on your topology. You should read about reflector peering and the implications of shared / unique cluster ids to make an informed decision.

For most use-cases it'll probably be unique to each Kubernetes cluster and could probably be set to any valid IPv4 address i.e. `1.1.1.1`.

Reply:
Noted.